controller/state.go

```diff
@@ -391,8 +391,14 @@ func (m *appStateManager) CompareAppState(app *v1alpha1.Application, project *ap
 		managedLiveObj = append(managedLiveObj, obj)
 	}
 
+	compareOptions, err := m.settingsMgr.GetResourceCompareOptions()
+	if err != nil {
+		log.Warnf("Could not get compare options from ConfigMap (assuming defaults): %v", err)
+		compareOptions = diff.GetDefaultDiffOptions()
```

Comment:
Since `CompareAppState` does not return faulty (no `error` return value), I deliberately chose to use default compare options if we can not get the options from the settings manager.